src/test/regress/sql/function_extensions.sql

```diff
@@ -417,3 +417,41 @@ insert into test_ao3 values(1,2),(2,3),(3,4);
 select count(*) from (select * from (select pg_catalog.gp_acquire_sample_rows('test_ao3'::regclass, 400, 'f')) ss limit 1) ss1;
 
 drop table test_ao3;
+
+-- Test initplan function referring the outer query.
+-- The query below previously led to a SIGSEGV.
+-- start_matchsubs
+-- m/ \(subselect\.c:.*\)/
+-- s/ \(subselect\.c:.*\)//
+-- end_matchsubs
+
+-- start_ignore
+drop table if exists test_table_cte;
+drop table if exists test_table;
+-- end_ignore
+
+create table test_table_cte(a text, b text);
+insert into test_table_cte values('-', '-');
+create table test_table(a int, b text);
+
+create or replace function test_function(param_text_in text, param_int_in int4) returns
+table (param_text_out text, param_int_out int4) as
+$function$
+begin
+return query
+with
+-- though cte1 is not used, we need it here to reproduce the SIGSEGV,
+-- without it the query below failed with 'invalid Datum pointer' error.
+cte1 as (select hostname from gp_segment_configuration),
+cte2 as (select a from test_table_cte where b = param_text_in)
+select param_text_in::text, param_int_in::int4 from cte2;
+end;
+$function$
+language plpgsql execute on initplan;
```

Comment:
You can simplify it a little like this
```suggestion
with
-- though cte1 is not used, we need it here to reproduce the SIGSEGV,
-- without it the query below failed with 'invalid Datum pointer' error.
cte1 as (select hostname from gp_segment_configuration),
cte2 as (select a from test_table_cte where b = param_text_in)
select param_text_in::text, param_int_in::int4 from cte2;
$function$
language sql execute on initplan;
```

Reply:
Ok